src/FlowtypeRunner.js

```diff
@@ -0,0 +1,48 @@
+const { exec } = require('child_process');
+const path = require('path');
+const { pass, fail } = require('create-jest-runner');
+
+class FlowtypeRunner {
+  constructor(globalConfig) {
+    this.globalConfig = globalConfig;
+  }
+
+  runTests(tests, watcher, onStart, onResult, onFailure, options) {
+    this.globalConfig = {};
+    const start = +new Date();
+    return new Promise((resolve) => {
+      exec('flow', { stdio: 'ignore', cwd: process.cwd() }, (err, stdout) => {
+        const errors = stdout.split('Error');
+        const errorsPerFile = errors.reduce((previous, current) => {
+          const firstErrorLine = current.split('\n')[0];
+          let fileName = firstErrorLine.split(':')[1];
+          if (fileName) {
+            fileName = path.join(process.cwd(), fileName.trim());
+            const errorMessage = current.substring(current.indexOf('\n') + 1);
+            if (!previous[fileName]) {
+              previous[fileName] = [];
+            }
+            previous[fileName].push(errorMessage);
+          }
+          return previous;
+        }, {});
+        tests.map((t) => {
+          if (!errorsPerFile[t.path]) {
+            const testResults = pass({ start, end: +new Date(), test: { path: t.path } });
+            onResult(t, testResults);
+          } else {
+            const testResults = fail({
+              start,
+              end: +new Date(),
+              test: { path: t.path, errorMessage: errorsPerFile[t.path] },
+            });
+            onResult(t, testResults);
+          }
+        });
+        resolve();
```

Comment:
I'm a little confused by resolving a promise & also returning a callback. Is there a reason you're doing both? Does the promise resolve in the event that nothing comes back from the callback?

Reply:
The resolve of the promise tells jest that the runner had finish his work. Calling the callbacks is telling jest that new test result is available. Actually those test results are working like generators where they yield new test results to jest when they are available.